app/helpers/application_helper.rb

```diff
@@ -150,16 +153,17 @@ def preload_script(script)
 
   def preload_script_url(url, entrypoint: nil)
     entrypoint_attribute = entrypoint ? "data-discourse-entrypoint=\"#{entrypoint}\"" : ""
+    nonce_attribute = "nonce=\"#{csp_nonce_placeholder}\""
```

Comment:
The nonce is present, but browsers hide it in dev-tools. I think because it's elided from the `outerHTML` API to avoid it being leaked.

If you access the element via JS and call `.nonce` then it'll show up. Or if you use 'view source' instead of inspect element.